lib/mp4/transmuxer.js

```diff
@@ -546,104 +547,170 @@ Transmuxer = function(options) {
+          coalesceStream.numberOfTracks++;
+          audioSegmentStream = new AudioSegmentStream(track);
+          // Set up the final part of the audio pipeline
+          adtsStream
+            .pipe(audioSegmentStream)
+            .pipe(coalesceStream);
+        }
+      }
+    });
+  };
 
-  // expose the metadata stream
-  this.metadataStream = new m2ts.MetadataStream();
-
-  options.metadataStream = this.metadataStream;
-
-  // set up the parsing pipeline
-  packetStream = new m2ts.TransportPacketStream();
-  parseStream = new m2ts.TransportParseStream();
-  elementaryStream = new m2ts.ElementaryStream();
-  aacStream = new AacStream();
-  h264Stream = new H264Stream();
-  captionStream = new m2ts.CaptionStream();
-  coalesceStream = new CoalesceStream(options);
-
-  // disassemble MPEG2-TS packets into elementary streams
-  packetStream
-    .pipe(parseStream)
-    .pipe(elementaryStream);
-
-  // !!THIS ORDER IS IMPORTANT!!
-  // demux the streams
-  elementaryStream
-    .pipe(h264Stream);
-  elementaryStream
-    .pipe(aacStream);
-
-  elementaryStream
-    .pipe(this.metadataStream)
-    .pipe(coalesceStream);
-
-  // Hook up CEA-608/708 caption stream
-  h264Stream.pipe(captionStream)
-    .pipe(coalesceStream);
+    this.setupTSPipeline = function() {
+    this.metadataStream = new m2ts.MetadataStream();
+
+    options.metadataStream = this.metadataStream;
+
+    // set up the parsing pipeline
+    packetStream = new m2ts.TransportPacketStream();
+    parseStream = new m2ts.TransportParseStream();
+    elementaryStream = new m2ts.ElementaryStream();
+    adtsStream = new AdtsStream();
+    h264Stream = new H264Stream();
+    captionStream = new m2ts.CaptionStream();
+    coalesceStream = new CoalesceStream(options);
+
+    // disassemble MPEG2-TS packets into elementary streams
+    packetStream
+      .pipe(parseStream)
+      .pipe(elementaryStream);
+
+    // !!THIS ORDER IS IMPORTANT!!
+    // demux the streams
+    elementaryStream
+      .pipe(h264Stream);
+    elementaryStream
+      .pipe(adtsStream);
+
+    elementaryStream
+      .pipe(this.metadataStream)
+      .pipe(coalesceStream);
+
+    // Hook up CEA-608/708 caption stream
+    h264Stream.pipe(captionStream)
+      .pipe(coalesceStream);
+    this.addTSListener();
+  };
 
-  // hook up the segment streams once track metadata is delivered
-  elementaryStream.on('data', function(data) {
-    var i;
-
-    if (data.type === 'metadata') {
-      i = data.tracks.length;
-
-      // scan the tracks listed in the metadata
-      while (i--) {
-        if (!videoTrack && data.tracks[i].type === 'video') {
-          videoTrack = data.tracks[i];
-          videoTrack.timelineStartInfo.baseMediaDecodeTime = self.baseMediaDecodeTime;
-        } else if (!audioTrack && data.tracks[i].type === 'audio') {
-          audioTrack = data.tracks[i];
-          audioTrack.timelineStartInfo.baseMediaDecodeTime = self.baseMediaDecodeTime;
+  this.addTSListener = function() {
```

Comment:
Our caps standard for this project means that this should be called `addTsListener`